tools/snippet-generator/Azure.Sdk.Tools.SnippetGenerator.Tests/TestData/Snippets.cs

```diff
@@ -0,0 +1,17 @@
+using NUnit.Framework;
+
+namespace Azure.Sdk.Tools.SnippetGenerator.TestData
+{
+    public class Snippets
+    {
+        [Test]
+        public void EmptySnippet()
+        {
+#if SHOULD_NEVER_BE_DEFINED
+            #region Snippet:EmptySnippet
+            throw new NotImplementedException();
```

Comment:
Curious: why is this single line considered empty?

Reply:
Because the symbol `SHOULD_NEVER_BE_DEFINED` wasn't defined, so there ends up being no content. This is effectively what happened in my case that prompted this feature, so I figured it would make a good test case.

Somehow (I didn't dig in too much), the region still gets defined. Presumably it's because of the visitor we have and how it's called by Roslyn, that all preprocessor instructions are considered "in".